vcenter_operator/configurator.py

```diff
@@ -316,21 +322,38 @@ def setup_os_session(self):
         )
         self.os_session = Session(auth=auth)
 
-    def _poll_nova(self):
-        if not self.os_session:
-            return
+    def _poll_nova_cells(self):
+        namespace_nova = self.global_options['namespace']
+        try:
+            secret = client.CoreV1Api().read_namespaced_secret(
+                namespace=namespace_nova,
+                name='nova-cells')
+        except client.ApiException as e:
+            LOG.error(f"Failed to get secret {namespace_nova}/nova-cells: {e}")
+            return False
+
+        if secret.type != 'Opaque':
+            LOG.error(f"Unknown secret type {secret.type} for {namespace_nova}/nova-cells. Can only work with Opaque.")
+            return False
 
         try:
-            endpoint_filter = {'service_type': 'compute', 'interface': 'public'}
-            resp = self.os_session.get('/os-cells', endpoint_filter=endpoint_filter)
-            for cell in resp.json().get('cellsv2', []):
-                self.global_options['cells'][cell['name']] = cell
-        except (HttpError, ConnectionError) as e:
-            LOG.error(f"Failed to get cells: {e}")
+            cell_names = b64decode(secret.data['cells']).split(',')
+            for cell_name in cell_names:
+                cell_data = {
+                    'database_connection': b64decode(secret.data[f"{cell_name}_database_connection"]),
+                    'transport_url': b64decode(secret.data[f"{cell_name}_transport_url"])}
```

Comment:
While not strictly necessary here, we probably expect these to be `str`-typed elswhere:
```suggestion
                    'database_connection': b64decode(secret.data[f"{cell_name}_database_connection"]).decode(),
                    'transport_url': b64decode(secret.data[f"{cell_name}_transport_url"]).decode()}
```